app/controllers/users.js

```diff
@@ -107,46 +112,125 @@ exports.create = function(req, res) {
   }
 };
 
+exports.signUp = (req, res) => {
+  if (req.body.name && req.body.password && req.body.email) {
+    User.findOne({
+      email: req.body.email
+    })
+      .exec((err, user) => {
+        if (!user) {
+          const newUser = new User(req.body);
+          newUser.avatar = avatars[newUser.avatar];
+          newUser.provider = 'local';
+          newUser.save((error) => {
+            if (error) {
+              return res.render('/#!/signup?error=unknown', {
+                errors: error.errors,
+                user: newUser
+              });
+            }
+
+            const { _id, email } = newUser;
+            const token = jwt.sign({
+              exp: Math.floor(Date.now() / 1000) + (60 * 60 * 24),
+              _id,
+              email
+            }, secret);
+            req.logIn(newUser, (err) => {
+              if (err) {
+                return res.status(500).send({ message: 'Internal Server Error' });
+              }
+              return res.status(201).send({
+                message: 'Signed up successfully',
+                token
+              });
+            });
+          });
+        } else {
+          return res.redirect('/#!/signup?error=existinguser');
+        }
+      });
+  } else {
+    return res.redirect('/#!/signup?error=incomplete');
+  }
+};
+
+exports.login = (req, res, next) => {
+  if (!req.body.email || !req.body.password) {
+    return res.status(406).json({
+      error: 'plaese fill in required fields'
+    });
+  }
+  return User.findOne({
+    email: req.body.email
+  }).exec((err, user) => {
+    if (!user) {
+      return res.status(401).json({
+        error: 'invalid credentials'
+      });
+    }
+    if (user && bcrypt.compareSync(req.body.password, user.hashed_password)) {
+      const {
+        _id,
+        email
+      } = user;
+      const token = jwt.sign({
+        exp: Math.floor(Date.now() / 1000) + (60 * 60 * 24),
+        _id,
+        email
+      }, secret);
+      req.logIn(user, (err) => {
+        if (err) return next(err);
+        return res.status(201).send({
```

Comment:
Hi, @Daramola98 status code used here on successful login shouldn't be `201`. Use `200 OK` instead